### PR TITLE
Disallow `patient_id` as a variable name

### LINKED ISSUE #1227: Add patient_id to the restricted variable names
`patient_id` should be a restricted name on a dataset:

Currently this dataset definition is valid:
```
from databuilder.ehrql import Dataset
from databuilder.tables.beta.tpp import patients

dataset = Dataset()
age = patients.age_on("2023-01-01")
dataset.define_population(age > 16)
dataset.age = age
dataset.patient_id = age
```
In locally generated output, patient_id has been overwritten with age (i.e. I get 2 identical columns)

### PR DESCRIPTION
**Pull request by evansd**
Fixes #1227

### CODE REVIEW

File: ehrql/query_language.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __setattr__(self, name, value):
             )
         if name in self.variables:
             raise AttributeError(f"'{name}' is already set and cannot be reassigned")
-        if name == "variables":
+        if name in ("patient_id", "variables"):
             raise AttributeError("'variables' is not an allowed variable name")
```
Comment (sebbacon) [inline ehrql/query_language.py:55]:
```suggestion
            raise AttributeError(f"'{name}' is not an allowed variable name")
```

Comment (iaindillingham) [reply ehrql/query_language.py:55]: Ah, you beat me to it!

**Review by iaindillingham — APPROVED** [review]
@sebbacon beat me to the f-string. Otherwise, LGTM.

File: ehrql/query_language.py (hunk 1 above)
Comment (evansd) [reply ehrql/query_language.py:55]: :facepalm: Thanks Seb (and Iain). It is very, very hot.

**Pull request merged**